WebRoot/js/ajax/dwt/xforms/XFormItem.js

```diff
@@ -995,10 +995,21 @@ XFormItem.prototype.handleKeyPressDelay = function(ev, domItem) {
 
 XFormItem.prototype.getKeyPressHandlerHTML = function () {
 
-	var keydownEv = "onkeydown";
-	if (AjxEnv.isNav) {
-		keydownEv = "onkeypress";
-	}
+        // if AjxEnv.isNav, etc.
+        var keydownEv = "onkeypress";
+
+        // IE, EDGE, & Firefox do not handle onkeypress consistenly
+        if (AjxEnv.isIE || AjxEnv.isIE3 || AjxEnv.isIE4 || AjxEnv.isIE4up ||
```

Comment:
I think (AjxEnv.isIE || AjxEnv.isModernIE)  should include mostly all versions of IE